code/assignTime.m

```diff
@@ -0,0 +1,279 @@
+        - dataTable.systemTick(iPacket - 1)), 2^16);
+end
+
+% Expected elapsed time for each packet, based on sample rate and number of
+% samples per packet; in units of systemTick (1e-4 seconds)
+expectedElapsed = dataTable.packetsizes .* (1./dataTable.samplerate) * 1e4;
+
+% If diff_systemTick and expectedElapsed differ by more than 10% of expectedElapsed,
+% flag as gap
+indices_systemTickFlagged = find (abs(expectedElapsed(2:end) - diff_systemTick(2:end)) > 0.1*expectedElapsed(2:end));
+
+% All packets flagged as end of continuous chunks
+allFlaggedIndices = unique([indices_changeFs; indices_timestampFlagged;...
+    indices_dataTypeSequenceFlagged; indices_systemTickFlagged],'sorted') ;
+
+%%
+disp('Chunking data')
+% Determine indices of packets which correspond to each data chunk
+if ~isempty(allFlaggedIndices)
+    counter = 1;
+    chunkIndices = cell(1,length(allFlaggedIndices) + 1);
+    for iChunk = 1:length(allFlaggedIndices)
+        if iChunk == 1
+            chunkIndices{counter} = (1:allFlaggedIndices(1));
+            currentStartIndex = 1;
+            counter = counter + 1;
+        elseif iChunk == length(allFlaggedIndices)
+            chunkIndices{counter} = allFlaggedIndices(currentStartIndex) + 1:allFlaggedIndices(currentStartIndex + 1);
+            chunkIndices{counter + 1} = allFlaggedIndices(currentStartIndex + 1) + 1:numPackets;
+        else
+            chunkIndices{counter} = allFlaggedIndices(currentStartIndex) + 1:allFlaggedIndices(currentStartIndex + 1);
+            currentStartIndex = currentStartIndex + 1;
+            counter = counter + 1;
+        end
+     end
+else
+    % No identified missing packets, all packets in one chunk
+    chunkIndices{1} = 1:numPackets;
+end
+
+%%
+% Loop through each chunk to determine offset to apply (as determined by
+% average difference between packetGenTime and expectedElapsed)
+disp('Determining start time of each chunk')
+
+% PacketGenTime in ms; convert difference to 1e-4 seconds, units of
+% systemTick and expectedElapsed
+diff_PacketGenTime = [1; diff(dataTable.PacketGenTime) * 1e1];
+
+numChunks = length(chunkIndices);
+chunksToExclude = [];
+meanError = NaN(1,numChunks);
+for iChunk = 1:numChunks
+    currentTimestampIndices = chunkIndices{iChunk};
+    
+    % Chunks must have at least 2 packets in order to have a valid
+    % diff_systemTick -- thus if chunk only one packet, it must be excluded
+    if length(currentTimestampIndices) == 1
+        chunksToExclude = [chunksToExclude iChunk];
+    end
+    % Always exclude the first packet of the chunk, because don't have an
+    % accurate diff_systemTick value for this first packet
+    currentTimestampIndices = currentTimestampIndices(2:end);
+    
+    % Differences between adjacent PacketGenTimes (in units of 1e-4
+    % seconds)
+    error = expectedElapsed(currentTimestampIndices) - diff_PacketGenTime(currentTimestampIndices);
+    meanError(iChunk) = median(error);
```

Comment:
should change to "medianError" so the variable is describing the operation.

Reply:
Resolved with f4c74f32f327afd99bbcda48173522199c42ab0c